File: 57. Insert Interval/insert.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> res;
        int index = 0, len = intervals.size();
        while (index < len && intervals[index][1] < newInterval[0])
            res.push_back(intervals[index++]);
        while (index < len && intervals[index][0] <= newInterval[1]) {
            newInterval = {min(intervals[index][0], newInterval[0]), max(intervals[index][1], newInterval[1])};
            index++;
        }
        res.push_back(newInterval);
        while (index < len)
            res.push_back(intervals[index++]);
        return res;
    }
};

File: 57. Insert Interval/insert.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        index = 0
        while index < len(intervals) and intervals[index][1] < newInterval[0]:
            res.append(intervals[index])
            index += 1
        while index < len(intervals) and intervals[index][1] < newInterval[0]:
            res.append(intervals[index])
            index += 1
        while index < len(intervals) and intervals[index][0] <= newInterval[1]:
            newInterval[0] = min(intervals[index][0], newInterval[0])
            newInterval[1] = max(intervals[index][1], newInterval[1])
            index += 1
        res.append(newInterval)
        while index < len(intervals):
            res.append(intervals[index])
            index += 1
        return res
            